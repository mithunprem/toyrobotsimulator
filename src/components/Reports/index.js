import React from 'react';
import CommandList from '../../Components/CommandList';
import RobotPositionReport from '../../Components/RobotPositionReport';
import PropTypes from 'prop-types';

const Reports = ({ reportRobotStatus, robotPosition, commandList }) => {

  return (
    <div className="reports m-3">
      { reportRobotStatus ?
        <RobotPositionReport robotPosition={robotPosition} /> : ''
      }
      { commandList.length > 0 ?
        <CommandList commandList={commandList} /> : ''
      }
    </div>
  );
}

Reports.propTypes = {
  reportRobotStatus: PropTypes.bool,
  robotPosition: PropTypes.object,
  commandList: PropTypes.array
}

export default Reports;
