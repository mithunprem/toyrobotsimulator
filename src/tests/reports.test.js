import React from 'react';
import { create } from 'react-test-renderer';
import Reports from '../components/reports';

describe("Reports", () => {
  const robotPosition = { x:1, y:1, direction:'NORTH' };
  const commandList = [ 'Place 0,0,north', 'move', 'report'];
  test("should render nothing on load of the page", () => {
    const report = create(
      <Reports commandList={[]}/>
  ).toJSON()
    expect(report).toMatchSnapshot()
  });

  test("should render the robotPosition when it is passed as a prop", () => {
    const report = create(
      <Reports
        commandList={[]}
        robotPosition={robotPosition}
        reportRobotStatus={true}/>
  ).toJSON()
    expect(report).toMatchSnapshot()
  });

  test("should render the commandList throwing an error", () => {
    const report = create(
      <Reports
        commandList={commandList} />
  ).toJSON()
    expect(report).toMatchSnapshot()
  });
});
