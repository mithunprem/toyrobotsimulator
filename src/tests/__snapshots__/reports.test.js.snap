// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports should render nothing on load of the page 1`] = `
<div
  className="reports m-3"
>
  
  
</div>
`;

exports[`Reports should render the commandList throwing an error 1`] = `
<div
  className="reports m-3"
>
  
  <div
    className="command-list"
  >
    <div
      className="command-list-header pl-3 pt-2 pb-2"
    >
      Commands executed
    </div>
    <table
      className="table m-0 table-sm table-striped table-bordered"
    >
      <tbody>
        <tr>
          <td
            className="p-2 command-row"
          >
            Place 0,0,north
          </td>
        </tr>
        <tr>
          <td
            className="p-2 command-row"
          >
            move
          </td>
        </tr>
        <tr>
          <td
            className="p-2 command-row"
          >
            report
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Reports should render the robotPosition when it is passed as a prop 1`] = `
<div
  className="reports m-3"
>
  <div
    className="report alert alert-info"
  >
    <span>
      Robot's current position is: 
      1
      , 
      1
      , 
      NORTH
    </span>
  </div>
  
</div>
`;
